src/Interpreters/Context.cpp

```diff
@@ -1526,6 +1528,17 @@ BackgroundSchedulePool & Context::getDistributedSchedulePool() const
     return *shared->distributed_schedule_pool;
 }
 
+BackgroundSchedulePool & Context::getMessageBrokerSchedulePool() const
+{
+    auto lock = getLock();
+    if (!shared->message_broker_schedule_pool)
+        shared->message_broker_schedule_pool.emplace(
+            settings.background_message_broker_schedule_pool_size,
```

Comment:
Does it work OK when it exceeds the `max_threads`?

Reply:
Sure, there is no such check for such settings